### PR TITLE
Feature/unbond unstake withdraw

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

> UI support for withdraw period. Existing call unbond_unstake_and_withdraw is splitted into two calls, one to start unbonding (unbond_and_unstake) and another to withdraw funds (withdraw_unbonded).

Don't merge until tested on Shibuya and runtime upgraded on Shiden

image

image

image

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes

**This pull request makes the following changes:**

**Adds**
- unbonded funds widget
- unlocking chunks dialog
- SSR support

**Fixes**
- unable to close register dialog

**Changes**
- unstake behavior

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,11 +0,0 @@
-const
-  express = require('express'),
```
Comment (hoonsubin) [inline server.js:2]: Since we're not going to be using SSR at the moment, can you revert this change? The SSR scripts can be kept, we just need this file for Heroku

Comment (bobo-kovacevic) [reply server.js:2]: Sure.

**Comment by hoonsubin** [comment]
just out of curiosity, why does the token symbol say `ASTL` instead of `ASTR`?

**Comment by bobo-kovacevic** [comment]
> 
That is a good question :) It looks like ASTL comes from a local node I am running
image

**Review by hoonsubin — APPROVED** [review]
~LGTM~

nvm found something new

File: src/components/dapp-staking/statistics/Withdraw.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    const unsub = subscribeToEraChange();
+
+    const getChunks = async (era: u32) => {
+      if (!canUnbondWithdraw) {
+        return;
+      }
+
+      const unbondingInfo =
+        await api?.value?.query.dappsStaking.unbondingInfoStorage<UnbondingInfo>(
```
Comment (hoonsubin) [inline src/components/dapp-staking/statistics/Withdraw.vue:95]: Screen Shot 2021-12-14 at 1 55 15 PM

The `unbondingInfoStorage` property is undefined on networks that does not yet implement the unbounding period feature on Astar network.

```
Uncaught (in promise) TypeError: _b.query.dappsStaking.unbondingInfoStorage is not a function
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:69)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:21)
    at new Promise (<anonymous>)
    at __awaiter (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:17)
    at getChunks (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:64)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:59)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///./node_modules/@quasar/app/lib/webpack/loader.js.transform-quasar-imports.js!./node_modules/ts-loader/index.js??clonedRuleSet-3.use[0]!./node_modules/@quasar/app/lib/webpack/loader.vue.auto-import-quasar.js??ruleSet[0].use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[1]!./src/components/dapp-staking/statistics/Withdraw.vue?vue&type=script&lang=ts:21)
    at new Promise (<anonymous>)
```

**Review by hoonsubin — CHANGES_REQUESTED** [review]
Some compatibility issues for the new dApps Staking pallet features

File: src/components/dapp-staking/statistics/Withdraw.vue (hunk 2 above)
Comment (bobo-kovacevic) [reply src/components/dapp-staking/statistics/Withdraw.vue:95]: Nice catch. I improved a logic for checking if unbonding is supported a little bit. Previously logic was checking a network name and now checks network feature to determine if unbonding is supported.

**Comment by hoonsubin** [comment]
Screen Shot 2021-12-20 at 1 33 59 PM
The loading takes too long that I get disconnected. Any ideas?

**Comment by bobo-k2** [comment]
> 

Seems to me that reward calculations causes this issue because number of eras is too high. I will take a look

**Review by hoonsubin — APPROVED** [review]
LGTM
I will merge this shortly
